### PR TITLE
Create PinCommand feature

### PR DESCRIPTION
**Pull request by cheahTJ**
Linked to #43, will work on the test cases in the upcoming commit

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/PinCommand.java [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import seedu.address.model.person.Person;
+
+
+/**
+ * Pins a person to the address book.
+ */
+public class PinCommand extends Command {
+    public static final String COMMAND_WORD = "pin";
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Pin a student to the top of the address book. "
+            + "Parameters: NusId (8 digits long, starting with an 'E'). \n"
```
Comment (Ragnapop) [inline src/main/java/seedu/address/logic/commands/PinCommand.java:22]: just realised nusid is 7 digits long, not 8

File: src/main/java/seedu/address/logic/commands/PinCommand.java [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    public PinCommand(NusId nusId) {
+        this.nusId = nusId;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+        Person personToPin = lastShownList.stream().filter(person -> person.getNusId().equals(nusId))
+                .findFirst().orElse(null);
```
Comment (Ragnapop) [inline src/main/java/seedu/address/logic/commands/PinCommand.java:39]: good use of streams

**Review by Ragnapop — COMMENTED** [review]
Overall, LGTM!

**Comment by hjuntan** [comment]
Require documentation of pin command: Can be added in another pr

**Pull request merged**